modules/calc/src/main/java/com/opengamma/strata/calc/config/ReportingCurrency.java

```diff
@@ -23,63 +24,85 @@
 import org.joda.beans.impl.direct.DirectMetaProperty;
 import org.joda.beans.impl.direct.DirectMetaPropertyMap;
 
-import com.opengamma.strata.basics.CalculationTarget;
 import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.calc.runner.function.CalculationFunction;
 
 /**
- * Reporting rules that always returns the same currency.
+ * The reporting currency.
  * <p>
- * These rules always return the same currency from {@link #reportingCurrency}.
+ * This is used to specify the currency that the result should be reporting in.
+ * The currency specified may be explicit, using {@link #of(Currency)}, or implicit
+ * using {@link #NATURAL}.
  */
 @BeanDefinition(builderScope = "private")
-final class FixedReportingRules implements ReportingRules, ImmutableBean {
+public final class ReportingCurrency
+    implements ImmutableBean, Serializable {
 
   /**
-   * The reporting currency that is used for every target.
+   * An instance requesting the "natural" currency of the target.
+   * <p>
+   * When converting calculation results, conversion will occur to the "natural" currency of the target.
+   * The "natural" currency of a target is obtained from {@link CalculationFunction}.
+   */
+  public static final ReportingCurrency NATURAL = new ReportingCurrency(ReportingCurrencyType.NATURAL, null);
+
+  /**
+   * The type of reporting currency.
    */
   @PropertyDefinition(validate = "notNull")
+  private final ReportingCurrencyType type;
```

Comment:
Without it, information is implied from the state without being explicit. We follow a similar pattern in some other places, where the type is not strictly necessary but adds clarity when observing the state (this isn't an case where every last piece of memory matters either). It also provides for future extensibility, should a need for PAY_LEG currency or similar be needed. While it is an extra class, the complexity is low and I consider the benefits to be well justified.